src/components/PhotoSwipe/InfoDialog/index.tsx

```diff
@@ -58,12 +58,28 @@ export function FileInfo({
     }, [metadata]);
 
     useEffect(() => {
-        if (!location && exif) {
-            const exifLocation = getEXIFLocation(exif);
-            if (exifLocation.latitude || exifLocation.latitude === 0) {
-                setLocation(exifLocation);
+        const main = async () => {
+            if (!location && exif) {
+                const exifLocation = getEXIFLocation(exif);
+                if (exifLocation.latitude || exifLocation.latitude === 0) {
+                    setLocation(exifLocation);
+                    try {
+                        let updatedFile = await updateFileLocation(
+                            currentItem,
+                            exifLocation
+                        );
+                        updatedFile = (
+                            await updateFilePublicMagicMetadata([updatedFile])
+                        )[0];
+                        updateExistingFilePubMetadata(currentItem, updatedFile);
+                        scheduleUpdate();
+                    } catch (e) {
+                        logError(e, 'failed to update file location');
+                    }
+                }
             }
-        }
+        };
```

Comment:
can we define this method outside `useEffect` and call it here? Also document the reason why we are doing this.. May be method name can be descriptive enough.. Something like `_updateFileMetadataWithExifLocationData`